File: apps/mobile/modules/clawdex-terminal/ios/ClawdexTerminalModule.swift
import ExpoModulesCore

public class ClawdexTerminalModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ClawdexTerminal")

    Function("getRendererInfo") {
      ClawdexGhosttyRuntime.rendererInfo.asDictionary()
    }

    View(ClawdexTerminalView.self) {
      Prop("sessionId") { (view: ClawdexTerminalView, sessionId: String?) in
        view.sessionId = sessionId
      }

      Prop("cols") { (view: ClawdexTerminalView, cols: Int) in
        view.setGrid(cols: cols, rows: view.rows)
      }

      Prop("rows") { (view: ClawdexTerminalView, rows: Int) in
        view.setGrid(cols: view.cols, rows: rows)
      }

      Prop("writeFrame") { (view: ClawdexTerminalView, frame: [String: Any]?) in
        view.applyWriteFrame(frame)
      }

      Prop("placeholderText") { (view: ClawdexTerminalView, placeholderText: String?) in
        view.placeholderText = placeholderText ?? view.placeholderText
      }

      Events("onReady", "onInput", "onTerminalResize")
    }
  }
}

File: apps/mobile/modules/clawdex-terminal/android/src/main/java/expo/modules/clawdexterminal/ClawdexTerminalModule.kt
package expo.modules.clawdexterminal

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ClawdexTerminalModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ClawdexTerminal")

    Function("getRendererInfo") {
      mapOf(
        "available" to false,
        "backend" to "android-placeholder",
        "message" to "Android is still using the placeholder renderer while iOS libghostty-vt work is wired up."
      )
    }

    View(ClawdexTerminalView::class) {
      Prop("sessionId") { view: ClawdexTerminalView, sessionId: String? ->
        view.sessionId = sessionId
      }

      Prop("cols") { view: ClawdexTerminalView, cols: Int ->
        view.setGrid(cols, view.rows)
      }

      Prop("rows") { view: ClawdexTerminalView, rows: Int ->
        view.setGrid(view.cols, rows)
      }

      Prop("writeFrame") { view: ClawdexTerminalView, frame: Map<String, Any?>? ->
        view.applyWriteFrame(frame)
      }

      Prop("placeholderText") { view: ClawdexTerminalView, placeholderText: String? ->
        view.placeholderText = placeholderText ?: view.placeholderText
      }

      Events("onReady", "onInput", "onTerminalResize")
    }
  }
}
